### PR TITLE
Add flag to fully disable compression for squashfs

### PR DESCRIPTION
**Pull request by Itxaka**
Signed-off-by: Itxaka <igarcia@suse.com>

### CODE REVIEW

File: pkg/action/build-iso.go [hunk 1]
```diff
@@ -145,11 +145,14 @@ func (b BuildISOAction) prepareISORoot(isoDir string, rootDir string, uefiDir st
 
 	b.cfg.Logger.Info("Creating squashfs...")
 	squashOptions := constants.GetDefaultSquashfsOptions()
```
Comment (davidcassany) [inline pkg/action/build-iso.go]: I think we could simply

```golang
squashOptions := append(constants.GetDefaultSquashfsOptions(), cfg.SquashFsCompressionConfig...)
```

File: pkg/elemental/elemental.go [hunk 2]
```diff
@@ -301,7 +301,10 @@ func (e *Elemental) DeployImage(img *v1.Image, leaveMounted bool) error {
 			return err
 		}
 		if img.FS == cnst.SquashFs {
-			opts := append(cnst.GetDefaultSquashfsOptions(), e.config.SquashFsCompressionConfig...)
+			opts := cnst.GetDefaultSquashfsOptions()
```
Comment (davidcassany) [inline pkg/elemental/elemental.go]: same here, we could slightly simplify it.

File: pkg/types/v1/config.go [hunk 3]
```diff
@@ -53,6 +53,7 @@ type Config struct {
 	Repos                     []Repository `yaml:"repositories,omitempty" mapstructure:"repositories"`
 	Arch                      string       `yaml:"arch,omitempty" mapstructure:"arch"`
 	SquashFsCompressionConfig []string     `yaml:"squash-compression,omitempty" mapstructure:"squash-compression"`
+	SquashFsNoCompression     bool         `yaml:"squash-no-compression,omitempty" mapstructure:"squash-no-compression"`
 }
 
 // Sanitize checks the consistency of the struct, returns error
```
Comment (davidcassany) [inline pkg/types/v1/config.go:59]: We could use the Sanitize method to ensure `SquashFsCompressionConfig` is an empty slice if `SquashFsNoCompression` is set. Then we can simply trust the config flags anywhere.

**Review by davidcassany — APPROVED** [review]
Nice, looks good :+1:

I think we can slightly simplify the former logic though. I realized the SquashFsCompressionConfig is already initialized in  `pkg/config/config.go` to the default. So we can omit the checks in `build-iso.go` and `elemental.go` and directly trust `v1.Config.SquashFsCompressionConfig` to include the custom values, defaults or nothing if `SquashFsNoCompression` was selected.

File: pkg/types/v1/config.go (hunk 3 above)
Comment (Itxaka) [reply pkg/types/v1/config.go:59]: ahh good idea!

**Pull request merged**